Source/WebCore/layout/formattingContexts/inline/InlineItemsBuilder.cpp

```diff
@@ -134,6 +135,60 @@ void InlineItemsBuilder::build(InlineItemPosition startPosition)
 #endif
 }
 
+void InlineItemsBuilder::computeInlineBoxBoundaryTextSpacingsIfNeeded()
+{
+    if (!m_hasTextAutospace)
+        return;
+
+    char32_t lastCharacterFromPreviousRun = 0;
+    size_t lastCharacterDepth = 0;
+    size_t currentCharacterDepth = 0;
+    InlineBoxBoundaryTextSpacings spacings;
+    Vector<unsigned> inlineBoxStartIndexesOnInlineItemsList;
+    bool processInlineBoxBoundary = false;
+    auto& inlineItemList = inlineContentCache().inlineItems().content();
+    for (unsigned inlineItemIndex = 0; inlineItemIndex < inlineItemList.size(); ++inlineItemIndex) {
+        auto& inlineItem = inlineItemList[inlineItemIndex];
+        if (inlineItem.isInlineBoxStart()) {
+            inlineBoxStartIndexesOnInlineItemsList.append(inlineItemIndex);
+            ++currentCharacterDepth;
+            continue;
+        }
+        if (inlineItem.isInlineBoxEnd()) {
+            --currentCharacterDepth;
+            processInlineBoxBoundary = true;
+            continue;
+        }
+        auto* inlineTextItem = dynamicDowncast<InlineTextItem>(inlineItem);
+        if (!inlineTextItem)
+            continue;
+
+        auto start = inlineTextItem->start();
+        auto length = inlineTextItem->length();
+        auto& inlineTextBox = inlineTextItem->inlineTextBox();
+        if (!processInlineBoxBoundary || !lastCharacterFromPreviousRun) {
+            lastCharacterFromPreviousRun = inlineTextBox.content().characterAt(start + length - 1);
+            lastCharacterDepth = currentCharacterDepth;
+            processInlineBoxBoundary = false;
+            continue;
+        }
+
+        unsigned boundaryDepth = std::min(currentCharacterDepth, lastCharacterDepth);
+        unsigned boundaryIndex = inlineBoxStartIndexesOnInlineItemsList.size() - 1 - (currentCharacterDepth - boundaryDepth);
+        ASSERT(boundaryIndex >= 0);
```

Comment:
This assert seems redundant. unsigned can't have negative value. if you wanna assert for arithmetic overflow, maybe assert on the  statement itself.